File: backend/app/schemas/question.py
from pydantic import BaseModel
from app.models.enums import DifficultyLevel


class QuestionBase(BaseModel):
    question_text: str
    option_a: str
    option_b: str
    option_c: str
    option_d: str
    difficulty: DifficultyLevel
    points: int


class QuestionCreate(QuestionBase):
    correct_option: str


class QuestionUpdate(QuestionCreate):
    pass


class QuestionResponse(QuestionBase):
    id: str

    class Config:
        from_attributes = True


class AdminQuestionResponse(QuestionResponse):
    correct_option: str

File: frontend/src/types/question.ts
export type Difficulty = "EASY" | "MEDIUM" | "HARD";
export type Option = "A" | "B" | "C" | "D";

export interface Question {
  id: string;
  question_text: string;
  option_a: string;
  option_b: string;
  option_c: string;
  option_d: string;
  difficulty: Difficulty;
  points: number;
}

export interface AdminQuestion extends Question {
  correct_option: Option;
}